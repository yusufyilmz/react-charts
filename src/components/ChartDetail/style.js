import styled from 'styled-components';

export const H2 = styled.h2`
    color: ${props => props.color};
    font-weight: bolder;
    float: ${props => props.index % 2 === 0 ? 'left' : 'right'}
`;

H2.displayName = 'h2'

export const Div = styled.div`
    display:flex;
    flex-direction:row;
    justify-content: space-between;
    float: ${props => props.index % 2 === 0 ? 'left' : 'right'}
`;

Div.displayName = 'div'

export const P = styled.p`
    margin: ${props => props.index % 2 === 0 ? '20px 20px 20px 0' : '20px 0 20px 20px'};
    color: ${props => props.type === 'value' && 'lightgray'};
`;

P.displayName = 'p'
