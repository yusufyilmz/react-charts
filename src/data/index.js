
export default [
    {
        title: 'REVENUE',
        totalValue: 200000,
        valueType: 'euro',
        values: [
            {
                name: 'Tablet',
                percentage: 60,
                color: '#47d147'
            },
            {
                name: 'Smartphone',
                percentage: 40,
                color: '#006600'
            }
        ],
        datas: [
            { id: 1, value: 1.1 },
            { id: 2, value: 2.2 },
            { id: 3, value: 1.8 },
            { id: 4, value: 3.2 },
            { id: 5, value: 2.5 },
            { id: 6, value: 5.2 },
            { id: 7, value: 4.6 },
            { id: 8, value: 6.3 },
            { id: 9, value: 4.8 },
        ]
    },
    {
        title: 'IMPRESSIONS',
        totalValue: 200000000,
        valueType: '',
        values: [
            {
                name: 'Tablet',
                percentage: 40,
                color: '#1ad1ff'
            },
            {
                name: 'Smartphone',
                percentage: 60,
                color: '#002699'
            }
        ],
        datas: [
            { id: 1, value: 5.2 },
            { id: 2, value: 4.8 },
            { id: 3, value: 5.0 },
            { id: 4, value: 3.8 },
            { id: 5, value: 2.3 },
            { id: 6, value: 2.8 },
            { id: 7, value: 1.5 },
            { id: 8, value: 2.5 },
            { id: 9, value: 2.0 },
        ]
    },
    {
        title: 'VISITS',
        totalValue: 600000000,
        valueType: '',
        values: [
            {
                name: 'Tablet',
                percentage: 80,
                color: '#ffcc00'
            },
            {
                name: 'Smartphone',
                percentage: 20,
                color: '#e68a00'
            }
        ],
        datas: [
            { id: 1, value: 1.1 },
            { id: 2, value: 2.2 },
            { id: 3, value: 1.8 },
            { id: 4, value: 3.2 },
            { id: 5, value: 2.5 },
            { id: 6, value: 5.2 },
            { id: 7, value: 4.6 },
            { id: 8, value: 6.3 },
            { id: 9, value: 4.8 },
        ]
    },
]

