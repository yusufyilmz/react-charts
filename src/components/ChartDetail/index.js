import React from 'react';
import { Div, H2, P} from './style'

const ChartDetail = ({ item, value, color, index }) => {

    return (
        <div>
            <H2 color={color} index={index} >{item.name}</H2>
            <Div index={index}>
                <P color={color} index={index}>{item.percentage}%</P>
                <P color={color} index={index} type="value">{value}</P>
            </Div>
        </div>
    )
}

export default ChartDetail;