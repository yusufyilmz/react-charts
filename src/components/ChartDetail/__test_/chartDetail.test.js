

import React from 'react';
import ChartDetail from '../index';
import { shallow } from 'enzyme'
import data from '../../../data';

describe('CHARTDETAIL COMPONENT', () => {

    let props, wrapper;

    beforeEach(() => {
        props = {
            item: data[0].values[0],
        }
        wrapper = shallow(
            <ChartDetail {...props} />
        );
    });

    test('renders element correctly', () => {
        expect(wrapper.find('div').length).toBe(2)
    });

    test('renders element correctly', () => {
        expect(wrapper.find('p').length).toBe(2)
    });

    test('renders element correctly', () => {
        expect(wrapper.find('h2').length).toBe(1)
    });

    test('renders text correctly', () => {
        expect(wrapper.find('h2').text()).toBe(props.item.name);
    });


    test('renders text correctly', () => {
        expect(wrapper.find('p').at(0).text()).toBe(props.item.percentage + '%');
    });

    test('snapchat correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

})
