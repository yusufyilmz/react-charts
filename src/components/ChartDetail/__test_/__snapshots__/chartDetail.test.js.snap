// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CHARTDETAIL COMPONENT snapchat correctly 1`] = `
<div>
  <h2>
    Tablet
  </h2>
  <div>
    <p>
      60
      %
    </p>
    <p
      type="value"
    />
  </div>
</div>
`;
